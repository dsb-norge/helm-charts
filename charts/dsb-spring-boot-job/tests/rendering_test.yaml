suite: test deployment
tests:
  - it: Minimal manifest should match snapshot
    templates:
      - cronJob.yaml
    set:
      image: nginx
      tag: latest
      schedule: "*/1 * * * *"
    asserts:
      - matchSnapshot: { }
  - it: Full manifest should match snapshot
    templates:
      - configMap.yaml
      - cronJob.yaml
      - databaseDeployment.yaml
      - databaseService.yaml
    set:
      replicas: 3
      minAvailableReplicas: 1
      image: wordpress
      tag: greatest
      imagePullPolicy: customValue
      cronJobAnnotations:
        key1: value1
        key2: value2
      springProfiles: "chart-testing"
      schedule: "*/2 * * * *"
      config:
        root:
          key1: value1
          key2: value2
      secretRefs:
        - secret1
      configMapRefs:
        - configMap1
      ingress_host: example.com
      ingress_path: /example
      application_web_port: 80
      application_actuator_port: 81
      prometheus_enabled: "false"
      prometheus_port: 82
      prometheus_path: "/path/to/prometheus"
      memory_request: "4Mi"
      memory_limit: "19Mi"
      cpu_request: "0.3"
      cpu_limit: "1"
      database_container:
        enabled: true
        image: "test-db:v1"
        database: "test-data"
        user: "custom user"
        password: "password123"
      jobs:
      - name: cron-example
        image_tag: greatest
        schedule: "*/2 * * * *"
        tokenURL: "https://example.com/token"
        apiPath: "/do"
        method: "TRACE"
      # Reference secrets that are mounted on the cron job pod:
      job_secretRefs:
        - jobSecret1
    asserts:
      - matchSnapshot: { }
