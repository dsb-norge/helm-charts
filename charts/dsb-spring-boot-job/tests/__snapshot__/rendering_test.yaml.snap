Full manifest should match snapshot:
  1: |
    apiVersion: v1
    data:
      root.key1: value1
      root.key2: value2
    kind: ConfigMap
    metadata:
      name: RELEASE-NAME-configmap
      namespace: NAMESPACE
  2: |
    apiVersion: batch/v1beta1
    kind: CronJob
    metadata:
      annotations:
        key1: value1
        key2: value2
      name: RELEASE-NAME
      namespace: NAMESPACE
    spec:
      concurrencyPolicy: Forbid
      failedJobsHistoryLimit: 3
      jobTemplate:
        spec:
          activeDeadlineSeconds: null
          backoffLimit: 3
          completions: 1
          parallelism: 1
          template:
            metadata:
              annotations:
                apparmor.security.beta.kubernetes.io/pod: runtime/default
                checksum: chart-version=2.0.4_config-hash=9ce8ae231891db9bc63c4cae273adbed21f7800c566a6dfaec7cddc5fea302a9
                co.elastic.logs/json.add_error_key: "true"
                co.elastic.logs/json.keys_under_root: "true"
                co.elastic.logs/json.message_key: message
                co.elastic.logs/json.overwrite_keys: "true"
              labels:
                app: RELEASE-NAME-job
            spec:
              containers:
              - env:
                - name: SPRING_PROFILES_ACTIVE
                  value: chart-testing
                - name: DATABASE_CONTAINER_USER
                  value: sa
                - name: DATABASE_CONTAINER_PASSWORD
                  value: password123
                - name: DATABASE_CONTAINER_HOST_AND_PORT
                  value: RELEASE-NAME-db-svc:1433
                - name: DATABASE_CONTAINER_DATABASE
                  value: emptydb
                envFrom:
                - secretRef:
                    name: secret1
                - configMapRef:
                    name: configMap1
                - configMapRef:
                    name: RELEASE-NAME-configmap
                image: wordpress:greatest
                name: RELEASE-NAME-job
                resources:
                  limits:
                    cpu: 1
                    memory: 19Mi
                  requests:
                    cpu: 0.3
                    memory: 4Mi
                securityContext:
                  allowPrivilegeEscalation: false
                  capabilities:
                    drop:
                    - ALL
                  privileged: false
                  readOnlyRootFilesystem: true
                  runAsGroup: 3000
                  runAsNonRoot: true
                  runAsUser: 1000
                volumeMounts:
                - mountPath: /tmp
                  name: tmp-dir
              nodeSelector:
                NodePool: workers
              restartPolicy: OnFailure
              volumes:
              - emptyDir: {}
                name: tmp-dir
      schedule: '*/2 * * * *'
      successfulJobsHistoryLimit: 2
  3: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: RELEASE-NAME-database-deployment
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app: RELEASE-NAME-db-app
      template:
        metadata:
          annotations:
            checksum: chart-version=2.0.4_config-hash=9ce8ae231891db9bc63c4cae273adbed21f7800c566a6dfaec7cddc5fea302a9
          labels:
            app: RELEASE-NAME-db-app
        spec:
          containers:
          - env:
            - name: ACCEPT_EULA
              value: "Y"
            - name: SA_PASSWORD
              value: password123
            - name: MSSQL_SA_PASSWORD
              value: password123
            image: test-db:v1
            name: RELEASE-NAME-database-container
            ports:
            - containerPort: 1433
            resources:
              limits:
                cpu: "2.5"
                memory: 2048Mi
              requests:
                cpu: "0.2"
                memory: 1024Mi
          hostname: RELEASE-NAME-db-svc
  4: |
    apiVersion: v1
    kind: Service
    metadata:
      name: RELEASE-NAME-db-svc
      namespace: NAMESPACE
    spec:
      ports:
      - port: 1433
        protocol: TCP
        targetPort: 1433
      selector:
        app: RELEASE-NAME-db-app
Minimal manifest should match snapshot:
  1: |
    apiVersion: batch/v1beta1
    kind: CronJob
    metadata:
      name: RELEASE-NAME
      namespace: NAMESPACE
    spec:
      concurrencyPolicy: Forbid
      failedJobsHistoryLimit: 3
      jobTemplate:
        spec:
          activeDeadlineSeconds: null
          backoffLimit: 3
          completions: 1
          parallelism: 1
          template:
            metadata:
              annotations:
                apparmor.security.beta.kubernetes.io/pod: runtime/default
                checksum: chart-version=2.0.4_config-hash=f15538db5d9d938e85141d4a8db9d69c8a129e36700fa3b50532b8828dcafd7b
                co.elastic.logs/json.add_error_key: "true"
                co.elastic.logs/json.keys_under_root: "true"
                co.elastic.logs/json.message_key: message
                co.elastic.logs/json.overwrite_keys: "true"
              labels:
                app: RELEASE-NAME-job
            spec:
              containers:
              - env:
                - name: SPRING_PROFILES_ACTIVE
                  value: kubernetes
                envFrom: null
                image: nginx:latest
                name: RELEASE-NAME-job
                resources:
                  limits:
                    cpu: 2.5
                    memory: 1024Mi
                  requests:
                    cpu: 0.2
                    memory: 256Mi
                securityContext:
                  allowPrivilegeEscalation: false
                  capabilities:
                    drop:
                    - ALL
                  privileged: false
                  readOnlyRootFilesystem: true
                  runAsGroup: 3000
                  runAsNonRoot: true
                  runAsUser: 1000
                volumeMounts:
                - mountPath: /tmp
                  name: tmp-dir
              nodeSelector:
                NodePool: workers
              restartPolicy: OnFailure
              volumes:
              - emptyDir: {}
                name: tmp-dir
      schedule: '*/1 * * * *'
      successfulJobsHistoryLimit: 2
